pennylane/data/utils.py

```diff
@@ -0,0 +1,346 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the Dataset utility functions.
```

Comment:
I thought we do not want a module named `utils`. Could this functions go to `dataset.py` or maybe a file with a more descriptive name? (not in this PR though)

Reply:
:+1:, we have been trying to remove existing `utils.py` files from around the codebase where possible

Reply:
I'll move things into dataset.py and keep it as clean as possible